language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -1,34 +1,44 @@
 -- | The CK machine.
 
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TemplateHaskell       #-}
 
 module Language.PlutusCore.Evaluation.Machine.Ck
     ( CkMachineException
+    , CkEvaluationException
     , EvaluationResult (..)
     , EvaluationResultDef
     , applyEvaluateCkBuiltinName
     , evaluateCk
     , runCk
     ) where
 
+import           PlutusPrelude
+
 import           Language.PlutusCore.Constant.Apply
 import           Language.PlutusCore.Core
 import           Language.PlutusCore.Evaluation.Machine.Exception
 import           Language.PlutusCore.Evaluation.Result
 import           Language.PlutusCore.Name
 import           Language.PlutusCore.View
-import           PlutusPrelude
-
-import           Data.Functor.Identity
 
 infix 4 |>, <|
 
 -- | The CK machine throws this error when it encounters a 'DynBuiltinName'.
-data NoDynamicBuiltinNamesMachineError = NoDynamicBuiltinNamesMachineError
+data NoDynamicBuiltinNamesMachineError
+    = NoDynamicBuiltinNamesMachineError
+    deriving (Show, Eq)
 
 -- | The CK machine-specific 'MachineException'.
 type CkMachineException = MachineException NoDynamicBuiltinNamesMachineError
 
+-- | The CK machine-specific 'EvaluationException'.
+type CkEvaluationException = EvaluationException NoDynamicBuiltinNamesMachineError ()
+
+type CkM = Either CkEvaluationException
```

Comment:
I recall we had a discussion about keeping the CK machine pure, but it turns out making it monadic doesn't really change anything (it was already monadic, in fact).